cmd/swarm/upload.go

```diff
@@ -36,22 +36,67 @@ import (
 	"gopkg.in/urfave/cli.v1"
 )
 
+var (
+	MAX_STDIN_SIZE = 1024 * 1000 * 100
+)
+
+func swapStdin() (size int, path string) {
+	b := make([]byte, 1024)
+	size = 0
+
+	td, err := ioutil.TempDir("", "swarm-stdin")
+	if err != nil {
+		utils.Fatalf("Could not create temporary directory: %s", err)
+	}
+	path = filepath.Join(td, "data")
+	f, err := os.Create(path)
+	if err != nil {
+		utils.Fatalf("Could not create temporary file: %s", err)
+	}
+	r := io.TeeReader(os.Stdin, f)
+
+	for true {
+		n, err := r.Read(b)
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			utils.Fatalf("STDIN Read failed: %s", err)
+		} else if n > MAX_STDIN_SIZE {
```

Comment:
I think you should use a `LimitReader`, but if you end up keeping this implementation, note that this check will always return false because `n` has a max value of `len(b)` (which is `1024`). You probably want to check `size` rather than `n`.